src/lib/dnssd/minimal_mdns/BUILD.gn

```diff
@@ -50,6 +53,22 @@ config("config") {
   }
 }
 
+source_set("policy_header") {
+  sources = [
+    "AddressPolicy.h",
+    "ServerIPAddresses.h",
+  ]
+}
+
+static_library("default_policy") {
+  sources = [ "AddressPolicy_DefaultImpl.cpp" ]
+
+  public_deps = [
+    ":policy_header",
+    "${chip_root}/src/inet",
+  ]
+}
```

Comment:
This seems to require the source be built  at the same time here, from within the chip checkout, even though the iterator is an interface.

It would be much more flexible to just define the interface in the header, but allow the implementers to pass/initialize the factory or iterator-getter in the initialization of minimal mdns server. That would also allow more easily configuring the object prior to its use, rather than expecting what amounts to a singular context-free global iterator

Reply:
Will update after side conversation about layout.

TLDR: will move policy as an abstract base class that can be injected.
Builds will be able to even exclude the default policy, to make sure their custom policy is actually enforced/used.

Reply:
Updated with abstract base class.